tests/test_model_method.py

```diff
@@ -0,0 +1,144 @@
+
+
+def approx(value, abs=0, rel=1e-6):
+    return pytest.approx(value, abs=abs, rel=rel)
+
+
+LOGGER = get_logger(__name__)
+
+
+@pytest.mark.parametrize(
+    'name, reference',
+    [
+        (
+            'cc',
+            {
+                'exc_order': None,
+                'ptb_order': None,
+                'slv': 'variational',
+                'corr': '',
+            },
+        ),
+        (
+            'ccd',
+            {
+                'exc_order': [2],
+                'ptb_order': None,
+                'slv': 'variational',
+                'corr': '',
+            },
+        ),
+        (
+            'ccsd',
+            {
+                'exc_order': [1, 2],
+                'ptb_order': None,
+                'slv': 'variational',
+                'corr': '',
+            },
+        ),
+        (
+            'ccsdt',
+            {
+                'exc_order': [1, 2, 3],
+                'ptb_order': None,
+                'slv': 'variational',
+                'corr': '',
+            },
+        ),
+        (
+            'ccsdtq',
+            {
+                'exc_order': [1, 2, 3, 4],
+                'ptb_order': None,
+                'slv': 'variational',
+                'corr': '',
+            },
+        ),
+        (
+            'ccsd(t)',
+            {
+                'exc_order': [1, 2],
+                'ptb_order': [3],
+                'slv': 'variational',
+                'corr': '',
+            },
+        ),
+        (
+            'ccsdt(q)',
+            {
+                'exc_order': [1, 2, 3],
+                'ptb_order': [4],
+                'slv': 'variational',
+                'corr': '',
+            },
+        ),
+        (
+            'qvccd',
+            {
+                'exc_order': [2],
+                'ptb_order': None,
+                'slv': 'quasi-variational',
+                'corr': '',
+            },
+        ),
+        (
+            'bccd',
+            {
+                'exc_order': [2],
+                'ptb_order': None,
+                'slv': 'Brueckner',
+                'corr': '',
+            },
+        ),
+        (
+            'ccd-f12',
+            {
+                'exc_order': [2],
+                'ptb_order': None,
+                'slv': 'variational',
+                'corr': 'F12',
+            },
+        ),
+        (
+            'ccd-r12',
+            {
+                'exc_order': [2],
+                'ptb_order': None,
+                'slv': 'variational',
+                'corr': 'R12',
+            },
+        ),
+    ],
+)
+def test_cc(name, reference):
+    method = CoupledCluster(type=name.upper())
+    method.normalize(None, LOGGER)
+    for ref, res in {'exc': 'excitation', 'ptb': 'perturbative'}.items():
+        res_order = getattr(method, f'{res}_order')
+        res_order = list(res_order) if res_order is not None else None
+        assert res_order == reference[f'{ref}_order']
```

Comment:
I would actually go testing individual methods instead of the full `normalize` function.